commands/commands.go

```diff
@@ -0,0 +1,186 @@
+	if res, err = h.Run(h.JsonCmd, w); err == nil {
+		if res, err = json.MarshalIndent(res, "", " "); err == nil {
+			fmt.Printf("%s\n", res)
+			return res, nil
+		}
+	}
+	return nil, err
+}
+
+var helper *JsonCmdHelper
+
+var preCfg *config.Config
+var fileCfg = config.Cfg
+
+func bindFlags() {
+	preCfg = &config.Config{}
+	RootCmd.PersistentFlags().StringVarP(&preCfg.ConfigFile, "configfile", "c", "config.toml", "config file")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.DebugLevel, "debuglevel", "d", "info", "Logging level {trace, debug, info, warn, error, critical}")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.AppDataDir, "appdatadir", "a", "", "wallet db path")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.LogDir, "logdir", "l", "", "log data path")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.Network, "network", "n", "testnet", "network")
+	RootCmd.PersistentFlags().BoolVar(&preCfg.Create, "create", false, "Create a new wallet")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.QServer, "qserver", "s", "127.0.0.1:8030", "qitmeer node server")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.QUser, "quser", "u", "admin", "qitmeer node user")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.QPass, "qpass", "p", "123456", "qitmeer node password")
+	RootCmd.PersistentFlags().StringVarP(&preCfg.WalletPass, "pubwalletpass", "P", "public", "data encryption password")
+	RootCmd.PersistentFlags().BoolVar(&preCfg.QNoTLS, "qnotls", fileCfg.QNoTLS, "disable TLS")
+	RootCmd.PersistentFlags().StringVar(&preCfg.QCert, "qcert", fileCfg.QCert, "Certificate path")
+	RootCmd.PersistentFlags().BoolVar(&preCfg.QTLSSkipVerify, "qtlsskipverify", fileCfg.QTLSSkipVerify, "skip TLS verification")
+
+	RootCmd.PersistentFlags().Int64Var(&preCfg.Confirmations, "confirmations", 10, "Number of block confirmations ")
+	RootCmd.PersistentFlags().BoolVar(&preCfg.UI, "ui", true, "Start Wallet with RPC and webUI interface")
+	RootCmd.PersistentFlags().StringArrayVar(&preCfg.Listeners, "listeners", fileCfg.Listeners, "rpc listens")
+	RootCmd.PersistentFlags().StringVar(&preCfg.RPCUser, "rpcUser", fileCfg.RPCUser, "rpc user,default by random")
+	RootCmd.PersistentFlags().StringVar(&preCfg.RPCPass, "rpcPass", fileCfg.RPCPass, "rpc pass,default by random")
+	RootCmd.PersistentFlags().Int64Var(&preCfg.MinTxFee, "mintxfee", fileCfg.MinTxFee, "The minimum transaction fee in QIT/kB default 20000 (aka. 0.0002 MEER/KB)")
+}
+
+// LoadConfig config file and flags
+func LoadConfig(cmd *cobra.Command, args []string) {
+	// load configfile ane merge command ,but don't udpate configfile
+	_, err := toml.DecodeFile(preCfg.ConfigFile, fileCfg)
+	if err != nil {
+		if cmd.Flag("configfile").Changed {
+			if fExit, _ := utils.FileExists(preCfg.ConfigFile); fExit {
+				log.Error(fmt.Sprintf("config file err: %s", err))
+				return
+			}
+			return
+		}
+	}
+
+	fileCfg.ConfigFile = preCfg.ConfigFile
+
+	if cmd.Flag("debuglevel").Changed {
+		fileCfg.DebugLevel = preCfg.DebugLevel
```

Comment:
The naming `fileCfg` and `preCfg` look like incorrect naming or lead to confusion. But we can improve/refactor later and let this RP go first.

Reply:
this function will be substituted by viper,  #43